numpy_groupies/aggregate_numpy.py

```diff
@@ -158,8 +158,7 @@ def _var(group_idx, a, size, fill_value, dtype=np.dtype(np.float64),
                           minlength=size) / (counts - ddof)
     if sqrt:
         ret = np.sqrt(ret)  # this is now std not var
-    if not np.isnan(fill_value):
```

Comment:
This seems like  an optimization gone wrong  i.e. it seems to assume that  the NaNs  are already in the  right  place.